keyring_darwin.go

```diff
@@ -47,15 +47,36 @@ func (k macOSXKeychain) Get(service, username string) (string, error) {
 	return strings.TrimSpace(fmt.Sprintf("%s", out)), nil
 }
 
-// Get gets a secret from the keyring given a service name and a user.
+// Set stores stores user and pass in the keyring under the defined service
+// name.
 func (k macOSXKeychain) Set(service, username, password string) error {
-	return exec.Command(
+	cmd := exec.Command(
 		execPathKeychain,
-		"add-generic-password",
-		"-U", //update if exists
-		"-s", service,
-		"-a", username,
-		"-w", password).Run()
+		"-i", // start in interactive mode to be able to pass the password using stdIn.
+	)
+
+	stdIn, err := cmd.StdinPipe()
+	if err != nil {
+		return err
+	}
+
+	err = cmd.Start()
+	if err != nil {
+		return err
+	}
+
+	// Write the command to stdIn.
+	command := fmt.Sprintf("add-generic-password -U -s %s -a %s -w %s\n", service, username, password)
```

Comment:
I'm not sure if there exists a Golang version of this but shellescape (https://github.com/xxorax/node-shell-escape/blob/master/shell-escape.js) is pretty robust in protection against injection attacks

Reply:
Drive-by comment: darwin keychain could probably be switched to use the DLL like [this](https://github.com/jaraco/keyring/blob/master/keyring/backends/_OS_X_API.py) python lib and there wouldn't be any worries about CLI usage.

Reply:
[keybase/go-keychain](https://github.com/keybase/go-keychain) uses cgo to communicate with the macOS Keychain API in the Security.framework. This also has the benefit of isolating keychain item access to the signed go binary using it (instead of the /usr/bin/security binary).

Reply:
This would also solve the referenced issue from kooky.

Reply:
> I'm not sure if there exists a Golang version of this but shellescape (https://github.com/xxorax/node-shell-escape/blob/master/shell-escape.js) is pretty robust in protection against injection attacks

Potentially something like [shellescape](https://github.com/alessio/shellescape) could work here.

Reply:
cgo here would make cross-compilation for darwin quite rough.

Reply:
> > I'm not sure if there exists a Golang version of this but shellescape (https://github.com/xxorax/node-shell-escape/blob/master/shell-escape.js) is pretty robust in protection against injection attacks
> 
> Potentially something like [shellescape](https://github.com/alessio/shellescape) could work here.

This is input for the security tool not a shell and would likely require a different kind of escaping. I can't test it without access to a macos machine though.